src/Renderer/RendererOpenGL.cpp

```diff
@@ -764,6 +764,21 @@ std::vector<NAS2D::DisplayDesc> NAS2D::RendererOpenGL::getDisplayModes() const
 	return result;
 }
 
+NAS2D::DisplayDesc NAS2D::RendererOpenGL::getClosestMatchingDisplayMode(const DisplayDesc& preferredDisplayDesc) const
+{
+	const auto display_index = SDL_GetWindowDisplayIndex(underlyingWindow);
+	SDL_DisplayMode preferred{};
+	preferred.w = preferredDisplayDesc.width;
+	preferred.h = preferredDisplayDesc.height;
+	preferred.refresh_rate = preferredDisplayDesc.refreshHz;
+
+	SDL_DisplayMode closest{};
+	if (SDL_GetClosestDisplayMode(display_index, &preferred, &closest))
+	{
+		return {closest.w, closest.h, closest.refresh_rate};
+	}
+	return {};
```

Comment:
Might be better to `throw` rather than silently return an unusable result.

Reply:
I was following what the underlying implementations do: SDL just returns a `nullptr` and DirectX returns an `HRESULT` denoting an error code. Neither throw.

Could document that if one isn't found a null description (all zeros) is returned.

Reply:
Neither of those can really throw. Exception handling with `throw` is C++ specific. Those libraries are fairly language agnostic.

It's up to the wrapper code to translate library error signalling to language specific error signalling.

Reply:
Makes sense.